android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/Utils.java

```diff
@@ -309,7 +310,8 @@ private static String cleanConditionString(String conditionStringRaw) {
         return conditionString.replaceAll("  ", " ");
     }
 
-    public static void buildRulesWithUniqueId(JSONObject element, String uniqueId, String ruleType, Context context, Map<String, List<Map<String, Object>>> rulesFileMap) throws JSONException {
+    public static void buildRulesWithUniqueId(JSONObject element, String uniqueId, String ruleType,
```

Comment:
Any reason for changing the method signature? This is a public method which could be called elsewhere.

Reply:
The step field is useful for it to be used elsewhere, on top of that that the step field is needed to ensure that only the fields that have that step prefix (address) will be converted during dynamic rules creating process

Reply:
Is this a bug fix? If not, a method can be added that has the same signature with the added `step` parameter and retain the original method which will now call the new method with a null `step` value. Then perform the new operations only if `step` is not null.

Reply:
yeah it is